File: archivist/src/test/java/com/zorroa/archivist/service/SharedLinkServiceTests.java
package com.zorroa.archivist.service;

import com.google.common.collect.ImmutableMap;
import com.google.common.collect.ImmutableSet;
import com.zorroa.archivist.AbstractTest;
import com.zorroa.archivist.domain.SharedLink;
import com.zorroa.archivist.domain.SharedLinkSpec;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import static com.zorroa.archivist.security.UtilsKt.getUserId;

/**
 * Created by chambers on 7/11/17.
 */
public class SharedLinkServiceTests extends AbstractTest {

    @Autowired
    TransactionEventManager transactionEventManager;

    @Autowired
    SharedLinkService sharedLinkService;

    @Test
    public void testSendEmail() throws InterruptedException {
        transactionEventManager.setImmediateMode(true);

        SharedLinkSpec spec = new SharedLinkSpec();
        spec.setSendEmail(true);
        spec.setState(ImmutableMap.of("foo", "bar"));
        spec.setUserIds(ImmutableSet.of(getUserId()));
        spec.setExpireTimeMs(1L);
        SharedLink link = sharedLinkService.create(spec);

        Thread.sleep(2000);
    }
}

File: archivist/src/test/kotlin/com/zorroa/archivist/service/SharedLinkServiceTests.kt
package com.zorroa.archivist.service

import com.zorroa.archivist.AbstractTest
import com.zorroa.archivist.domain.SharedLinkSpec
import com.zorroa.archivist.security.getUserId
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired

/**
 * Created by chambers on 7/11/17.
 */
class SharedLinkServiceTests : AbstractTest() {

    @Autowired
    lateinit var sharedLinkService: SharedLinkService

    @Test
    @Throws(InterruptedException::class)
    fun testSendEmail() {
        transactionEventManager.isImmediateMode = true

        val spec = SharedLinkSpec(mapOf("foo" to "bar"))
        spec.userIds = setOf(getUserId())
        sharedLinkService.create(spec)
    }
}
